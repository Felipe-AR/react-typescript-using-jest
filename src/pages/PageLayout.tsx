import Card from "../components/Card"
import Formulario from "../components/Form"
import ListaParticipantes from "../components/ParticipantList"
import Rodape from "../components/Footer"

const PageLayout = () => {
    return (
        <Card>
            <section>
                <h2>Vamos come√ßar!</h2>
                <Formulario />
                <ListaParticipantes />
                <Rodape />
            </section>
        </Card>
    )
}

export default PageLayout