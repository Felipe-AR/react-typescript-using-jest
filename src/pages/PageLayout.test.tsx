import { render } from "@testing-library/react";
import React from "react";
import { RecoilRoot } from "recoil";
import PageLayout from "./PageLayout";

const mockNavegacao = jest.fn()

jest.mock('react-router-dom', () => {
    return {
        useNavigate: () => mockNavegacao
    }
})

describe('a pagina de configuracao', () => {
    test('deve ser renderizada corretamente', () => {
        const { container } = render(<RecoilRoot>
            <PageLayout />
        </RecoilRoot>)

        expect(container).toMatchSnapshot()
    })
})