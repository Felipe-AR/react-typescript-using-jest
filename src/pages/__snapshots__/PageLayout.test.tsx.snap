// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a pagina de configuracao deve ser renderizada corretamente 1`] = `
<div>
  <div
    class="sc-bczRLJ eYAlEs"
  >
    <section>
      <h2>
        Vamos come√ßar!
      </h2>
      <form>
        <div
          class="sc-gsnTZi ioSZGc"
        >
          <input
            class="sc-dkzDqf gHLDox"
            placeholder="Insira os nomes dos participantes"
            type="text"
            value=""
          />
          <button
            class="sc-hKMtZM dOWzhi"
            disabled=""
          >
            Adicionar
          </button>
        </div>
        
      </form>
      <ul />
      <footer
        class="sc-jSMfEi gfNvPW rodape-configuracoes"
      >
        <button
          class="sc-gKXOVf bUxDgU botao"
          disabled=""
        >
          Iniciar brincadeira
        </button>
        <img
          alt="Sacolas de compras"
          class="sc-iBkjds gUwjft"
          src="/imagens/sacolas.png"
        />
      </footer>
    </section>
  </div>
</div>
`;
